app/src/main/java/com/grarak/kerneladiutor/fragments/other/SettingsFragment.java

```diff
@@ -324,7 +324,7 @@ private void dumpsysfs() {
         // loop through each array in the constants file. These contain all the other arrays.
         // have to do this once for the 1d arrays and again for the 2 arrays
         try {
-            File sysfsdump = new File(MainActivity.context.getFilesDir(), "sysfsdump.txt");
+            File sysfsdump = File.createTempFile("sysfsdump", ".txt", MainActivity.context.getExternalMediaDirs()[0]);
```

Comment:
I shall amend it to call deleteOnExit then. The problem with getCacheDir(which I did try) is that several email apps will only allow files from an external dir such as /sdcard. Gmail for instance has this limitation.